nixos/modules/services/x11/desktop-managers/cinnamon.nix

```diff
@@ -0,0 +1,155 @@
+
+       ${cfg.extraGSettingsOverrides}
+     EOF
+
+     ${pkgs.glib.dev}/bin/glib-compile-schemas $out/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas/
+    '';
+in
+
+{
+
+  options = {
+    services.xserver.desktopManager.cinnamon = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable the Cinnamon desktop environment.";
+      };
+
+      extraGSettingsOverrides = mkOption {
+        default = "";
+        type = types.lines;
+        description = "Additional gsettings overrides.";
+      };
+
+      extraGSettingsOverridePackages = mkOption {
+        default = [];
+        type = types.listOf types.path;
+        description = "List of packages for which gsettings are overridden.";
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    # Enable colord server
+    services.colord.enable = mkDefault true;
+
+    # Enable dconf
+    programs.dconf.enable = mkDefault true;
+
+    # Enable org.a11y.Bus
+    services.gnome3.at-spi2-core.enable = true;
+
+    # Fix lockscreen
+    security.pam.services = {
+      cinnamon-screensaver = {};
+    };
+
+    # Power managment support
+    services.upower.enable = mkDefault true;
+
+    # Network manager
+    networking.networkmanager.enable = mkDefault true;
+
+    # Override GSettings schemas
+    environment.sessionVariables.NIX_GSETTINGS_OVERRIDES_DIR = "${nixos-gsettings-desktop-schemas}/share/gsettings-schemas/nixos-gsettings-overrides/glib-2.0/schemas";
+
+    environment.systemPackages = with pkgs.cinnamon // pkgs; [
+      # common-files
+      cinnamon-common
+      cinnamon-session
+      cinnamon-desktop
+
+      # utils needed by some scripts
+      pkgs.killall
+
+      # session requirements
+      cinnamon-screensaver
+      # nemo-autostart: provided by nemo
+      pkgs.gnome3.networkmanagerapplet
+      # cinnamon-killer-daemon: provided by cinnamon-common
+
+      # packages
+      nemo
+      cinnamon-control-center
+      cinnamon-settings-daemon
+      gnome3.libgnomekbd
+      # TODO: these are optional, but part of "the cinnamon experience" - add via option like gnome? (also not packaged yet)
+      # warp
+      # blueberry
+
+      # theme
+      pkgs.gnome3.adwaita-icon-theme
+      pkgs.hicolor-icon-theme
+      pkgs.gnome3.gnome-themes-extra
+      pkgs.gtk3.out # for gtk-launch
+      mint-themes
+      mint-y-icons
+
+      # other
+      pkgs.glib # for gsettings
+      pkgs.shared-mime-info # for update-mime-database
+      pkgs.xdg-user-dirs
+    ];
+
+    environment.pathsToLink = [
+      "/share" # TODO: https://github.com/NixOS/nixpkgs/issues/47173
+    ];
+
+    fonts.fonts = with pkgs; [
+      cantarell-fonts
+      dejavu_fonts
+      source-code-pro # Default monospace font in 3.32
+      source-sans-pro
+    ];
+
```

Comment:
you should be able to replace these both with
```
 services.xserver.displayManager.sessionPackages = [ 
  cinnamon.cinnamon-common
];
```

however, the xsession files aren't patched properly
```
[Desktop Entry]
Name=Cinnamon
Comment=This session logs you into Cinnamon
Exec=cinnamon-session-cinnamon
TryExec=/usr/bin/cinnamon
Icon=
Type=Application
```

The stream of execution requires patching like
1. the xsession files to`$out/bin`
2. $out/bin/cinnamon-session-cinnamon and $out/bin/cinnamon-session-cinnamon2d 
```
# these scripts need to be hardcoded to ${cinnamon-session}/bin/cinnamon-session
exec cinnamon-session --session cinnamon "$@"
```